Packs/FeedRecordedFuture/Integrations/FeedRecordedFuture/FeedRecordedFuture.py

```diff
@@ -159,21 +158,76 @@ def build_iterator(self, service, indicator_type, risk_rule: Optional[str] = Non
                              " requests made to Recorded Future. ")
             else:
                 return_error(
-                    '{} - exception in request: {} {}'
-                    .format(self.SOURCE_NAME, response.status_code, response.content))  # type: ignore
+                    f'{self.SOURCE_NAME} - exception in request: {response.status_code} {response.content}'  # type: ignore
+                )
 
         if service == 'connectApi':
-            response_content = gzip.decompress(response.content)
-            response_content = response_content.decode('utf-8')
-            with open("response.txt", "w") as f:
-                f.write(response_content)
+            self.stream_compressed_data(response=response, chunk_size=CHUNK_SIZE)
         else:
+            demisto.debug("Will now stream the response's data")
             with open("response.txt", "w") as f:
                 for chunk in response.iter_content(CHUNK_SIZE, decode_unicode=True):
                     if chunk:
                         f.write(chunk)
         demisto.info('done build_iterator')
 
+    def stream_compressed_data(self, response: requests.Response, chunk_size: int) -> None:
+        """This will stream the response's compressed data and write it to a file. The streamed data is compressed,
+        therefore, for every chunk that we stream, we will decode it, but we may decode A PART of a character, for which
+        the decoder will throw an error.
+        We solve this by removing one byte form the chunk and decode again, until we decode
+        successfully, and the bytes that were removed will be concatenated to the next chunk.
+
+        Args:
+            response (requests.Response): The response object.
+            chunk_size (int): The chunk size to use while streaming the data.
+
+        Raises:
+            DemistoException: If we reach a point where we need to cut more than 3 bytes from the chunk.
+        """
+        demisto.debug("Will now stream the response's compressed data")
+        # Since we need to decompress gzip compressed data, we add 16 to zlib.MAX_WBITS, to tell the decompressor object
+        # that we are dealing with gzip
+        decompressor = zlib.decompressobj(zlib.MAX_WBITS + 16)
+        cut_off_bytes = b''
+        chunks_counter = 0
+        decoded_counter = 0
+        with open("response.txt", "w") as f:
+            for chunk in response.iter_content(chunk_size):
+                if chunk:
+                    chunks_counter += 1
+                    decompressed_chunk = decompressor.decompress(chunk)
+                    if cut_off_bytes:
+                        # To concatenate cut off bytes from previous chunk
+                        decompressed_chunk = cut_off_bytes + decompressed_chunk
+                        cut_off_bytes = b''
+                    chunk_to_decode = decompressed_chunk
+                    for bytes_to_cut in [1, 2, 3, 4]:
```

Comment:
```suggestion
                    for bytes_to_cut in [0, 1, 2, 3]:
```
first one should not remove bytes if not needed.